tests/ops/qubit/test_non_parametric_ops.py

```diff
@@ -613,12 +611,31 @@ def test_invalid_mixed_polarity_controls(
             ([1, 2, 0], 3, "100"),
             ([1, 0, 2, 4], 3, "1001"),
             ([0, 1, 2, 5, 3, 6], 4, "100001"),
+            (None,[0, 1], "0"),
+            (None,[0, 1, 2], "00"),
+            (None,[0, 1, 2], "10"),
+            (None,[1, 0, 2], "10"),
+            (None,[0, 1, 2], "11"),
+            (None,[0, 2, 1], "10"),
+            (None,[1, 2, 0, 3], "100"),
+            (None,[1, 0, 2, 4, 3], "1001"),
+            (None,[0, 1, 2, 5, 3, 6, 4], "100001"),
         ],
     )
     def test_mixed_polarity_controls(self, control_wires, wires, control_values):
```

Comment:
How about we instead make two copies of this test: `test_mixed_polarity_controls` and `test_mixed_polarity_controls_old`.  Then when we deprecated the `control_wires` keyword, we can just delete one entire test.

We also want to explicitly catch the Deprecation Warnings with [`pytest.warns`](https://docs.pytest.org/en/6.2.x/warnings.html#warns).  We want to make sure warnings are raised when we expect them to be. We also don't want unexpected warnings at the bottom of the pytest report, as they can indicate potential bugs and things going wrong. Currently, our test suite raises a bunch of warnings, but I'm trying to fix that.